code/modules/mob/dead/new_player/new_player.dm

```diff
@@ -161,6 +162,12 @@
 				to_chat(usr, "<span class='warning'>Server is full.</span>")
 				return
 
+		// Check if random role is requested
+		if(href_list["SelectedJob"] == "Random")
+			var/datum/job/job = SSjob.GetRandomJob(src)
+			if(job != FALSE)
```

Comment:
You need to abort if the job is false as it's going to try spawning the person with the role of "Random"

Add an if(!job)
                     Return

Preferably with some error message to the user

Reply:
True. I figured I wasn't that needed as the AssignRole() will also abort the job assignment with an error. I can see about a new message to returning there.

Reply:
Fixed